classes/item.rb

```diff
@@ -0,0 +1,20 @@
+class Item
+  attr_accessor :id, :genre, :author, :label, :publish_date, :archive
+
+  def initialize(genre:, author:, label:, publish_date:)
+    @id = Random.rand(1..10_000)
+    @genre = genre
+    @author = author
+    @label = label
+    @publish_date = publish_date
+    @archive = false
+  end
+
+  def move_to_archive
+    @archive = true if can_be_archived?
+  end
+
+  def can_be_archived?
+    @publish_date > 10
+  end
+end
```

Comment:
- [ ] We are missing a very important part of the requirement that says `All properties visible in the diagram should be defined and set up in the constructor method. **Exception: properties for the 1-to-many relationships should NOT be set in the constructor method. Instead, they should have a custom setter method created**.` So let's create setters for all the 1-to-many relationships `@genre`, `@author`, `@label` instead of setting them up in the constructor method. Check out this [article](http://ruby-for-beginners.rubymonstas.org/writing_classes/attribute_writers.html) to learn how to do this. See hint below:

```ruby
def genre=(genre)
      @genre = genre
      genre.items.push(self) unless genre.items.include?(self)
  end
```